jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletHolder.java

```diff
@@ -747,11 +747,14 @@ protected void initMultiPart () throws Exception
         //cleaned up correctly
         if (((Registration)getRegistration()).getMultipartConfig() != null)
         {
+            if (LOG.isDebugEnabled())
+                LOG.debug("multipart cleanup listener added for {}", this);
+
             //Register a listener to delete tmp files that are created as a result of this
             //servlet calling Request.getPart() or Request.getParts()
-
             ContextHandler ch = ContextHandler.getContextHandler(getServletHandler().getServletContext());
-            ch.addEventListener(MultiPartCleanerListener.INSTANCE);
+            if(!Arrays.asList(ch.getEventListeners()).contains(MultiPartCleanerListener.INSTANCE))
```

Comment:
ServletHolder.initServlet() should only be called once - do you have a repro test case where it is called multiple times?

Reply:
I don't currently have a test for this, Greg seemed to think that this could be called for different servlets using the same ContextHandler.

Reply:
You (and Greg) are perfectly correct - without this test multiple servlets that support multipart could add the instance multiple times.